addons/payment_xendit/controllers/main.py

```diff
@@ -58,3 +58,12 @@ def _verify_notification_token(self, received_token, tx_sudo):
         if not consteq(tx_sudo.provider_id.xendit_webhook_token, received_token):
             _logger.warning("Received notification with invalid callback token %r.", received_token)
             raise Forbidden()
+
+    @http.route('/payment/xendit/payment', type='json', auth='public')
+    def xendit_payment(self, reference, partner_id, token_id):
+        """ Take newly created token, do a charge request and handle the response
+
+        :param str reference: The reference of the transation
+        """
+        tx_sudo = request.env['payment.transaction'].sudo().search([('reference', '=', reference)])
+        tx_sudo._xendit_create_charge(token_id)
```

Comment:
I'm a bit concerned by the security aspect of this call. Basically, if an attacker finds someone else's token ID, they can use it to pay for their own transaction that was created beforehand.

Reply:
`_xendit_create_charge` calls `_xendit_make_request`  which will still require the `xendit_secret_key` field when calling the API . So I think that attacker would only be successful when he knows both the token and also the secret key. 
If the secret key is already exposed, I think attacker would pretty much be able to do anything :laughing:

Reply:
They can call the route with an arbitrary reference and token ID, so they would actually only need to know the latter. I will check with the security team.

Reply:
@odoo/rd-security could we have your input here?

Reply:
@xmo-odoo thank you for your input here! Indeed, Xendit allows a single-use token as well (make the is_multiple_use: False)

In this case, I've made the change such that single/multi-use is conditional on whether user wants to save the payment method. That's what I'm getting from your feedback, but  please do correct me if I'm wrong or if there's still anything I miss!
cc @AntoineVDV